pulp_container/app/serializers.py

```diff
@@ -696,3 +717,72 @@ class Meta:
             "tag",
             "artifacts",
         )
+
+
+class RepositorySignSerializer(serializers.Serializer):
+    """
+    Serializer for container images signing.
+    """
+
+    manifest_signing_service = serializers.HyperlinkedRelatedField(
+        required=False,
+        view_name="signing-services-detail",
+        queryset=models.ManifestSigningService.objects.all(),
+        help_text=_(
+            "A signign service to sign with. This will override a singing service set on the repo."
+        ),
+    )
+
+    # Ask for the future_base_path for synced repos - this should match future/existing distribution
+    # otherwise client verification can fail, it looks at 'docker-reference' in the signature json
+    future_base_path = serializers.CharField(
```

Comment:
While a push-repo has always associated a distribution to it and we can retrieve this information from distribution.base_path, a synced repo may not be distributed at this moment. We need to resort to the user's input to the extent where if bad path will be provided, later on podman will fail verification in case `future_base_path!= created later ditribution.base_path`